internal/download/get.go

```diff
@@ -77,30 +77,31 @@ func httpGetWithProgressRetry(url string, progress *progress.Progress, attempt i
 		}
 	}
 
-	bar := progress.AddByteProgressBar(int64(total))
-
-	// Ensure bar is always closed (especially for retries)
-	defer func() {
-		if !bar.Completed() {
-			bar.Abort(true)
-		}
-	}()
-
 	src := resp.Body
 	var dst bytes.Buffer
-
-	src = bar.ProxyReader(resp.Body)
+	var bar *progress.TotalBar
+	if prog != nil {
+		bar = prog.AddByteProgressBar(int64(total))
+		src = bar.ProxyReader(resp.Body)
+
+		// Ensure bar is always closed (especially for retries)
+		defer func() {
+			if !bar.Completed() {
+				bar.Abort(true)
+			}
+		}()
+	}
 
 	_, err = io.Copy(&dst, src)
 	if err != nil {
 		logging.Debug("Reading body failed: %s", err)
 		if attempt <= retries {
-			return httpGetWithProgressRetry(url, progress, attempt+1, retries)
+			return httpGetWithProgressRetry(url, prog, attempt+1, retries)
```

Comment:
I wonder how the progress bar actually behaves for retries here.  The deferred `bar.Abort()` call will not have been issued at this point.  So, the (not really) aborted `ByteProgressBar` that we added above is still active somehow.  I suppose it is a little too much for this PR (which isn't about this anyways), but maybe we could have a follow-up story to investigate this...

Reply:
Good point, if this is an issue then it existed before the changes here and we haven't seen any reports or noticed any issues with the progress bar on retries. I agree we should investigate to be sure and potentially update this function to use our retry library.

Story is here: https://www.pivotaltracker.com/story/show/176683389